awscli/topics/config-vars.rst

```diff
@@ -0,0 +1,209 @@
+:title: AWS CLI Configuration Variables
+:description: Configuration Variables for the AWS CLI
+:category: General
+:related command: configure, configure get, configure set
+
+Configuration values for the AWS CLI can come from several sources:
+
+* As a command line option
+* As an environment variable
+* As a value in the AWS CLI config file
+* As a value in the AWS Shared Credential file
+
+Some options are only available in the AWS CLI config.  This topic guide covers
+all the configuration variables available in the AWS CLI.
+
+Note that if you are just looking to get the minimum required configuration to
+run the AWS CLI, we recommend running ``aws configure``, which will prompt you
+for the necessary configuration values.
+
+Config File Format
+==================
+
+The AWS CLI config file, which defaults to ``~/.aws/config`` has the following
+format::
+
+    [default]
+    aws_access_key_id=foo
+    aws_secret_access_key=bar
+    region=us-west-2
+
+The ``default`` section refers to the configuration values for the default
+profile.  You can create profiles, which represent logical groups of
+configuration.  Profiles that aren't the default profile are specified by
+creating a section titled "profile profilename"::
+
+    [profile testing]
+    aws_access_key_id=foo
+    aws_secret_access_key=bar
+    region=us-west-2
+
+Nested Values
+-------------
+
+Some service specific configuration, discussed in more detail below, has a
+single top level key, with nested sub values.  These sub values are denoted by
+indentation::
+
+    [profile testing]
+    aws_access_key_id = foo
+    aws_secret_access_key = bar
+    region = us-west-2
+    s3 =
+      max_concurrent_requests=10
+      max_queue_size=1000
+
+
+General Options
+===============
+
+The AWS CLI has a few general options:
+
+=========== ========= ===================== ===================== ============================
+Variable    Option    Config Entry          Environment Variable  Description
+=========== ========= ===================== ===================== ============================
+profile     --profile profile               AWS_DEFAULT_PROFILE   Default profile name
```

Comment:
Profile's entry in the config is different than region and output, which could be confusing. You may want to make it so that it shows up exactly as it would in the config file. So profile would be ``[profile <value>]`` and region would be ``region=<value>`` etc. Then you could also show how the option on the command line maps to the config file by appending ``<value>`` to each option so make it ``--profile <value>`` and do the same for environment variables possibly for consistency.